packages/xforms-engine/src/parse/model/TranslationDefinitionMap.ts

```diff
@@ -0,0 +1,23 @@
+import type { DOMItextTranslationElement } from '../XFormDOM.ts';
+
+const generateChunksForLanguage = (
+	translationElement: DOMItextTranslationElement
+): Map<string, Element> => {
+	return new Map(
+		Array.from(translationElement.children).map((textElement) => {
+			const itextId = textElement.getAttribute('id');
+			return [itextId!, textElement] as const;
+		})
+	);
+};
+
+export class TranslationDefinitionMap extends Map<string, Map<string, Element>> {
+	constructor(translationElements: readonly DOMItextTranslationElement[]) {
+		super(
+			translationElements.map((translationElement) => {
+				const lang = translationElement.getAttribute('lang');
+				return [lang, generateChunksForLanguage(translationElement)] as const;
+			})
+		);
+	}
+}
```

Comment:
This file was renamed because now it's a lot simpler. I tried removing this altogether but it turned out that it was expensive to look the element up by ID so keeping the map sped up looking up the element which more than made up for the initial load.